cmd/fs-v1-multipart.go

```diff
@@ -474,7 +605,22 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 		return "", err
 	}
 
-	uploadIDPath := path.Join(bucket, object, uploadID)
+	bucketDir, err := fs.getBucketDir(bucket)
+	if err != nil {
+		return "", toObjectErr(err, bucket)
+	}
+
+	if _, err = fsStatDir(bucketDir); err != nil {
+		return "", toObjectErr(traceError(err), bucket)
+	}
+
+	uploadIDPath := pathJoin(bucket, object, uploadID)
+
+	rwlk, err := fs.rwPool.Create(pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket, object, uploadsJSONFile))
+	if err != nil {
+		return "", toObjectErr(traceError(err), bucket, object)
+	}
+	defer rwlk.Close()
```

Comment:
Ah that way. Let me see if there are more places which doesn't follow this sequence.